controllers/articlesController.js

```diff
@@ -93,4 +82,154 @@ export default class PostController {
       });
     }
   }
+
+  /**
+   * @description It helps the user to fetch all of the created articles.
+   * @param  {object} req - The request object
+   * @param  {object} res - The response object
+   * @returns {object} It returns the request's response object
+   */
+
+  static async findAll(req, res, next) {
+    const { user: { id = null } = {} } = req;
+    const { author, favorited, tag } = req.query;
+    const { page = 1, limit = 20 } = req.query;
```

Comment:
@Musigwa Line 96 can be included in line 95 to avoid duplication

Reply:
I'm not disagreeing with you, but the more properties being destructured from the object on a single line, the more the code formatter nests them onto the new lines which ultimately creates a destructuring hell of nested objects/properties which results in maintainability bad practices. To make it clean, I'm destructing a few properties on line 96 and the others on 95. I might be wrong but, I think there is no duplication from that unless you have some convincing technical reasons why we shouldn't do it like that.

Reply:
I meant having it
```
const {
 author,
 favorited,
 tag,
 page = 1,
 limit = 20
} = req.query;
```
instead of 
```
const { author, favorited, tag } = req.query;
const { page = 1, limit = 20 } = req.query;
```
You are destructuring two times from req.query, yet it can be done in a single line and also I did get what you meant by nested objects/properties in line 95 and 96

Reply:
Okay, I now got the point. I'm sorry for I was getting exactly what you mean. Let me apply it as soon as possible and I'll ping when I'm done.